engine/src/main/java/org/terasology/logic/players/PlayerSystem.java

```diff
@@ -187,14 +205,36 @@ public void onRespawnRequest(RespawnRequestEvent event, EntityRef entity) {
         entity.saveComponent(loc);
 
         if (worldProvider.isBlockRelevant(spawnPosition)) {
-            spawnPlayer(entity);
+            respawnPlayer(entity);
         } else {
             updateRelevanceEntity(entity, ViewDistance.LEGALLY_BLIND.getChunkDistance());
             SpawningClientInfo info = new SpawningClientInfo(entity, spawnPosition);
-            clientsPreparingToSpawn.add(info);
+            clientsPreparingToRespawn.add(info);
         }
     }
 
+    private void respawnPlayer(EntityRef clientEntity) {
+
+        ClientComponent client = clientEntity.getComponent(ClientComponent.class);
+        EntityRef playerCharacter = client.character;
+        LocationComponent location = clientEntity.getComponent(LocationComponent.class);
+        PlayerFactory playerFactory = new PlayerFactory(entityManager, worldProvider);
+        Vector3f spawnPosition = playerFactory.findSpawnPositionFromLocationComponent(location);
+        location.setWorldPosition(spawnPosition);
+        clientEntity.saveComponent(location);
+
+        AliveCharacterComponent aliveCharacterComponent = new AliveCharacterComponent();
+        playerCharacter.addComponent(aliveCharacterComponent);
```

Comment:
Could just short-cut these two lines and `addComponent(new AliveCharacterComponent():`